### ISSUE TITLE
[Editor] Context menu `Add conditional breakpoints` should focus condition input

### ISSUE THREAD

**Issue by digitarald** [devtools-html/debugger.html#4823]
STR:
- Open https://mozilladevelopers.github.io/sample-todo/03-conditional-breakpoints/
- Right click on a line number in Debugger and click `Add conditional breakpoints`

AR:
- Input isn't focused

ER:
- Focus input field to start typing condition

This works as expected when the breakpoint is created normally and then made conditional by right clicking `Add condition` in the `Breakpoints` section.

**Comment by GarethSharpe** [comment]
I wasn't able to reproduce the issue. Possible to send a gif of the AR?

![field-focus](https://user-images.githubusercontent.com/22062107/34470903-a1ed6a60-ef09-11e7-8386-8ca9afb9eca1.gif)

**Comment by digitarald** [comment]
Not working for me in Nightly/OSX using the STR.

**Comment by AlexisDeschamps** [comment]
I was not able to reproduce on Nightly and Chrome on Windows 10. Could this be specific to OSX?

**Comment by jasonLaster** [comment]
@AlexisDeschamps yeah - it could be...

**Comment by AlexisDeschamps** [comment]
It seems that the problem arises when the Debugger is used within a panel instead of just the launchpad.

**Comment by mmcote** [comment]
Found that the .focus() is correctly setting itself below in the renderToWidget and am currently searching for where it gets switched to focus elsewhere.

**Comment by mmcote** [comment]
Doing this simple check:

`
const activeElementIsFocused =
document.activeElement == this.input && document.hasFocus();
console.log("Input is focused:", activeElementIsFocused);
`

Right after the this.input.focus() in the renderToWidget it will always return true whether the conditional panel input is focused or not. Whereas, if you add in a componentDidUpdate section and check under that lifecycle event this will accurately return true when the input box shows focused and false when the input box is not focused. 

This tells me that somewhere in between renderToWidget and componentDidUpdate the input loses focus.. sometimes.

Originally I fixed the issue with a setTimeout in the renderToWidget, thereafter I changed the fix to just put the this.input.focus() into the componentDidUpdate.

Currently, I am trying to write a mochi test for the issue.
